modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java

```diff
@@ -1543,4 +1543,67 @@ public void shouldGenerateBeanValidationOnHeaderParams() throws IOException {
                 .containsWithNameAndAttributes("Min", ImmutableMap.of("value", "500"))
                 .containsWithNameAndAttributes("Max", ImmutableMap.of("value", "10000"));
     }
+
+    @Test
+    public void shouldUseEqualsNullableForArrayWhenSetInConfig_issue13385() throws IOException {
+        File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
+        output.deleteOnExit();
+
+        OpenAPI openAPI = new OpenAPIParser()
+                .readLocation("src/test/resources/bugs/issue_13385.yml", null, new ParseOptions()).getOpenAPI();
+        SpringCodegen codegen = new SpringCodegen();
+        codegen.setLibrary(SPRING_BOOT);
+        codegen.setOutputDir(output.getAbsolutePath());
+        codegen.additionalProperties().put(SpringCodegen.INTERFACE_ONLY, "true");
+        codegen.additionalProperties().put(SpringCodegen.USE_BEANVALIDATION, "true");
+        codegen.additionalProperties().put(SpringCodegen.PERFORM_BEANVALIDATION, "true");
+        codegen.additionalProperties().put(CodegenConstants.MODEL_PACKAGE, "xyz.model");
+        codegen.additionalProperties().put(CodegenConstants.API_PACKAGE, "xyz.controller");
+
+        ClientOptInput input = new ClientOptInput()
+                .openAPI(openAPI)
+                .config(codegen);
+
+        DefaultGenerator generator = new DefaultGenerator();
+        Map<String, File> files = generator.opts(input).generate().stream()
+                .collect(Collectors.toMap(File::getName, Function.identity()));
+
+        JavaFileAssert.assertThat(files.get("TestObject.java"))
+                .printFileContent()
+                .assertMethod("equals")
+                .bodyContainsLines("return equalsNullable(this.picture, testObject.picture);");
+
+        JavaFileAssert.assertThat(files.get("TestObject.java")).printFileContent()
+                .hasImports("com.fasterxml.jackson.annotation.JsonIgnore");
+
+    }
+      @Test
+      public void shouldNotUseEqualsNullableForArrayWhenNotSetInConfig_issue13385() throws IOException {
```

Comment:
Nit: indentation looks off

Reply:
Hi, this has been resolved in latest commit